papermill/cli.py

```diff
@@ -88,26 +88,39 @@ def _resolve_type(value):
         return False
     elif value == "None":
         return None
-    elif _is_float(value):
-        return float(value)
     elif _is_int(value):
         return int(value)
+    elif _is_float(value):
+        return float(value)
     else:
         return value
 
 
 def _is_int(value):
-    if type(value) is int:
-        return True
-    else:
+    """
+    Casting is used to check if the value
+    in the cli can be converted to an int
+    """
+    try:
+        float_cast = float(value)
+        int_cast = int(float_cast)
```

Comment:
Instead of doing this here, I'd keep the ordering of where `_is_float` is in the `_resolve_type` check.